Akopova 02 20

francoisa-inga opened the pull request:
Error messages counter added

SeleniumQALight commented on akopova/src/main/java/pages/LoginPage.java:
```diff
@@ -63,6 +93,58 @@ public void enterPasswordSignIn(String password) {
     }
 
+    public LoginPage enterLoginSignUp(String login){
+        try {
```
try у нас уже есть в методе enterTextIntoElement , так что он не нужен тут

francoisa-inga replied:
Done

SeleniumQALight commented on akopova/src/main/java/pages/LoginPage.java:
```diff
@@ -63,6 +93,58 @@ public void enterPasswordSignIn(String password) {
+
+
+    public LoginPage enterEmailSignUp(String eMail){
+        try {
```
try у нас уже есть в методе enterTextIntoElement , так что он не нужен тут

francoisa-inga replied:
Done

SeleniumQALight commented on akopova/src/main/java/pages/LoginPage.java:
```diff
@@ -63,6 +93,58 @@ public void enterPasswordSignIn(String password) {
+    }
+
+    public LoginPage enterPasswordSignUp(String password){
+        try {
```
try у нас уже есть в методе enterTextIntoElement , так что он не нужен тут

francoisa-inga replied:
Done

SeleniumQALight commented on akopova/src/main/java/pages/LoginPage.java:
```diff
@@ -63,6 +93,58 @@ public void enterPasswordSignIn(String password) {
+    }
+
+    public LoginPage enterDataForSignUp(WebElement element, String text){
+        try {
```
try у нас уже есть в методе enterTextIntoElement , так что он не нужен тут

francoisa-inga replied:
Done

SeleniumQALight commented on akopova/src/main/java/pages/LoginPage.java:
```diff
@@ -63,6 +93,58 @@ public void enterPasswordSignIn(String password) {
+
+    public LoginPage enterDataForSignUp(WebElement element, String text){
+        try {
+            element.clear();
```
очистка элемента есть в enterTextIntoElement, так что тут она лишняя

francoisa-inga replied:
Done

SeleniumQALight commented on akopova/src/main/java/pages/LoginPage.java:
```diff
@@ -71,11 +153,26 @@ public void clickButtonSignIn() {
+    }
+
+    public void clickButtonSignUp() {
+        try{
```
try у нас уже есть в методе clickOnElement, так что он не нужен тут

SeleniumQALight commented on akopova/src/main/java/pages/LoginPage.java:
```diff
@@ -90,4 +187,115 @@ public HomePage loginWithValidCreds() {
+        MyUtil.waitABit(5);
+        clickButtonSignUp();
+        // Check that we registered correct User : Homepage should open
+        MyUtil.waitABit(10);
```
нужно использовать явное ожидание вместо  MyUtil.waitABit(10);

SeleniumQALight commented on akopova/src/main/java/pages/LoginPage.java:
```diff
@@ -90,4 +187,115 @@ public HomePage loginWithValidCreds() {
+
+    public HomePage registerNewValidUser() {
+        enterValidSignUpInfo();
+        MyUtil.waitABit(5);
```
нужно использовать явное ожидание вместо  MyUtil.waitABit(5);

SeleniumQALight commented on akopova/src/main/java/pages/LoginPage.java:
```diff
@@ -90,4 +187,115 @@ public HomePage loginWithValidCreds() {
+
+
+
+    public LoginPage enterInvalidSignUpInfo(String signUpLoginValue
```
а если я буду вводить етим методом валидные значение он разве не введет? Думаю что можно назвать его enteSignUpInfo

francoisa-inga replied:
Done

SeleniumQALight commented on akopova/src/main/java/pages/LoginPage.java:
```diff
@@ -90,4 +187,115 @@ public HomePage loginWithValidCreds() {
+    }
+
+    public HomePage checkIsRedirectedOnHomePage(){
+
```
ту нужно добавить ожидания скрытия Чата

francoisa-inga replied:
Done

SeleniumQALight commented on akopova/src/main/java/resources/config.properties:
```diff
@@ -1,4 +1,5 @@
 base_url=https://qa-complex-app-for-testing.herokuapp.com
+homePage_url
```
это часть HomePage, его сюда не нужно выносить

francoisa-inga replied:
Removed

SeleniumQALight commented on akopova/src/test/java/userRegistrationTests/RegisterNewUserTest.java:
```diff
@@ -0,0 +1,75 @@
+    @Test
+    public void registerValidUser() {
+        //loginPage.openLoginPage();
+        loginPage.registerNewValidUser();
```
в этом тесте нет проверки. Вернее она то есть внутри данного метода, но должна быть в тесте .

francoisa-inga replied:
Done

SeleniumQALight commented on akopova/src/test/java/userRegistrationTests/RegisterNewUserTest.java:
```diff
@@ -0,0 +1,75 @@
+    }
+*/
+    @Test
+    public void checkErrorMessagesAmounts() {
```
нужно добавить параметризацию , если это уже домашка с занятия 25.02

francoisa-inga replied:
// TODO after merging the previous homework

SeleniumQALight reviewed (commented):
Поправь по комментам и на каждый что то отпишись.

francoisa-inga commented:
Ответила ниже по каждому
    On Friday, February 26, 2021, 01:41:35 PM GMT+2, SeleniumQALight <notifications@github.com> wrote:  
 
 


@SeleniumQALight commented on this pull request.

Поправь по комментам и на каждый что то отпишись.

In akopova/src/main/java/pages/LoginPage.java:
> @@ -63,6 +93,58 @@ public void enterPasswordSignIn(String password) {
         enterTextIntoElement(inputPassword, password);
     }
 
+    public LoginPage enterLoginSignUp(String login){
+        try {
try у нас уже есть в методе enterTextIntoElement , так что он не нужен тут
Done
In akopova/src/main/java/pages/LoginPage.java:
> +    public LoginPage enterLoginSignUp(String login){
+        try {
+            enterTextIntoElement(signUpLogin, login);
+        }
+        catch (Exception e){
+            logger.error(e + "login signup exception");
+        }
+
+        return this;
+    }
+
+
+
+
+    public LoginPage enterEmailSignUp(String eMail){
+        try {
try у нас уже есть в методе enterTextIntoElement , так что он не нужен тут
Done
In akopova/src/main/java/pages/LoginPage.java:
> +
+
+
+    public LoginPage enterEmailSignUp(String eMail){
+        try {
+            enterTextIntoElement(signUpEmail, eMail);
+        }
+        catch (Exception e){
+            logger.error(e + "Email signup exception");
+        }
+
+        return this;
+    }
+
+    public LoginPage enterPasswordSignUp(String password){
+        try {
try у нас уже есть в методе enterTextIntoElement , так что он не нужен тут
Done
In akopova/src/main/java/pages/LoginPage.java:
> +        return this;
+    }
+
+    public LoginPage enterPasswordSignUp(String password){
+        try {
+            enterTextIntoElement(signUpPassword, password);
+        }
+        catch (Exception e){
+            logger.error(e + "Email signup exception");
+        }
+
+        return this;
+    }
+
+    public LoginPage enterDataForSignUp(WebElement element, String text){
+        try {
try у нас уже есть в методе enterTextIntoElement , так что он не нужен тут
Done
In akopova/src/main/java/pages/LoginPage.java:
> +    }
+
+    public LoginPage enterPasswordSignUp(String password){
+        try {
+            enterTextIntoElement(signUpPassword, password);
+        }
+        catch (Exception e){
+            logger.error(e + "Email signup exception");
+        }
+
+        return this;
+    }
+
+    public LoginPage enterDataForSignUp(WebElement element, String text){
+        try {
+            element.clear();
очистка элемента есть в enterTextIntoElement, так что тут она лишняя
Done
In akopova/src/main/java/pages/LoginPage.java:
> @@ -71,11 +153,26 @@ public void clickButtonSignIn() {
         }
         catch (Exception e) {
             logger.error("Can't work with SignIn Button");
-            Assert.fail("Can't work with SignIn Button");
+            fail("Can't work with SignIn Button");
+
+        }
+    }
+
+    public void clickButtonSignUp() {
+        try{
try у нас уже есть в методе clickOnElement, так что он не нужен тут
Done
In akopova/src/main/java/pages/LoginPage.java:
> @@ -90,4 +187,115 @@ public HomePage loginWithValidCreds() {
         return new HomePage(webDriver);
     }
 
+    public void enterValidSignUpInfo() {
+        openLoginPage();
+        enterDataForSignUp(signUpLogin, TestData.VALID_SIGNUP_LOGIN);
+        enterDataForSignUp(signUpEmail, TestData.VALID_SIGNUP_EMAIL);
+        enterDataForSignUp(signUpPassword, TestData.VALID_SIGNUP_PASSWORD);
+
+    }
+
+    public HomePage registerNewValidUser() {
+        enterValidSignUpInfo();
+        MyUtil.waitABit(5);
+        clickButtonSignUp();
+        // Check that we registered correct User : Homepage should open
+        MyUtil.waitABit(10);
нужно использовать явное ожидание вместо MyUtil.waitABit(10);
Тут мы решили его пока оставить.
In akopova/src/main/java/pages/LoginPage.java:
> @@ -90,4 +187,115 @@ public HomePage loginWithValidCreds() {
         return new HomePage(webDriver);
     }
 
+    public void enterValidSignUpInfo() {
+        openLoginPage();
+        enterDataForSignUp(signUpLogin, TestData.VALID_SIGNUP_LOGIN);
+        enterDataForSignUp(signUpEmail, TestData.VALID_SIGNUP_EMAIL);
+        enterDataForSignUp(signUpPassword, TestData.VALID_SIGNUP_PASSWORD);
+
+    }
+
+    public HomePage registerNewValidUser() {
+        enterValidSignUpInfo();
+        MyUtil.waitABit(5);
нужно использовать явное ожидание вместо MyUtil.waitABit(5);
Пока оставляем, буду инвестигейтить
In akopova/src/main/java/pages/LoginPage.java:
> +
+    }
+
+    public HomePage registerNewValidUser() {
+        enterValidSignUpInfo();
+        MyUtil.waitABit(5);
+        clickButtonSignUp();
+        // Check that we registered correct User : Homepage should open
+        MyUtil.waitABit(10);
+        checkIsRedirectedOnHomePage();
+        return new HomePage(webDriver);
+    }
+
+
+
+    public LoginPage enterInvalidSignUpInfo(String signUpLoginValue
а если я буду вводить етим методом валидные значение он разве не введет? Думаю что можно назвать его enteSignUpInfo
Done
In akopova/src/main/java/pages/LoginPage.java:
> +
+    protected LoginPage waitForErrorMessageToBecomeVisible(String errorMessageLocator){
+        webDriverWait10.until(ExpectedConditions
+                .visibilityOfElementLocated(By.xpath(errorMessageLocator)));
+        return this;
+    }
+
+    public HomePage waitForHomePageToLoad() {
+        webDriverWait10.until(ExpectedConditions
+                .visibilityOfElementLocated(By.xpath(".//*[@class='header-bar mb-3']")));
+        return new HomePage(webDriver);
+
+    }
+
+    public HomePage checkIsRedirectedOnHomePage(){
+
ту нужно добавить ожидания скрытия Чата
Done
In akopova/src/main/java/resources/config.properties:
> @@ -1,4 +1,5 @@
 base_url=https://qa-complex-app-for-testing.herokuapp.com
+homePage_url
это часть HomePage, его сюда не нужно выносить
Removed
In akopova/src/test/java/userRegistrationTests/RegisterNewUserTest.java:
> @@ -0,0 +1,75 @@
+package userRegistrationTests;
+
+import baseTest.BaseTest;
+import libs.MyUtil;
+import libs.TestData;
+import org.junit.Test;
+
+public class RegisterNewUserTest extends BaseTest {
+
+
+    @Test
+    public void registerValidUser() {
+        //loginPage.openLoginPage();
+        loginPage.registerNewValidUser();
в этом тесте нет проверки. Вернее она то есть внутри данного метода, но должна быть в тесте .
Added
In akopova/src/test/java/userRegistrationTests/RegisterNewUserTest.java:
> +                .checkErrorMessagesAmount(loginPage.signUpErrorMessageLocator, 2)
+        ;
+
+        loginPage.signUpEmail.clear();
+        loginPage.enterEmailSignUp(TestData.VALID_SIGNUP_EMAIL);
+        MyUtil.waitABit(5);
+        loginPage.checkErrorMessagesAmount(loginPage.signUpErrorMessageLocator, 3);
+
+        loginPage.signUpPassword.clear();
+        loginPage.enterPasswordSignUp(TestData.VALID_PASSWORD);
+        MyUtil.waitABit(5);
+        loginPage.checkErrorMessagesAmount(loginPage.signUpErrorMessageLocator, 2);
+    }
+*/
+    @Test
+    public void checkErrorMessagesAmounts() {
нужно добавить параметризацию , если это уже домашка с занятия 25.02
// TODO
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

francoisa-inga commented on akopova/src/main/java/pages/LoginPage.java:
```diff
@@ -35,47 +67,67 @@ public void openLoginPage() {
+        return this;
     }
 
+    public LoginPage enterDataForSignUp(WebElement element, String text) {
```
Done

francoisa-inga commented on akopova/src/main/java/pages/LoginPage.java:
```diff
@@ -90,4 +142,117 @@ public HomePage loginWithValidCreds() {
 
+    public void enterValidSignUpInfo() {
+        openLoginPage();
+        enterDataForSignUp(signUpLogin
```
Done

francoisa-inga commented on akopova/src/main/java/pages/LoginPage.java:
```diff
@@ -90,4 +142,117 @@ public HomePage loginWithValidCreds() {
+        openLoginPage();
+        enterDataForSignUp(signUpLogin
+                , TestData.VALID_SIGNUP_LOGIN + MyUtil.getDateAndTimeFormated());
+        enterDataForSignUp(signUpEmail, TestData.VALID_SIGNUP_EMAIL);
```
Done

francoisa-inga commented on akopova/src/main/java/pages/LoginPage.java:
```diff
@@ -90,4 +142,117 @@ public HomePage loginWithValidCreds() {
+        enterDataForSignUp(signUpLogin
+                , TestData.VALID_SIGNUP_LOGIN + MyUtil.getDateAndTimeFormated());
+        enterDataForSignUp(signUpEmail, TestData.VALID_SIGNUP_EMAIL);
+        enterDataForSignUp(signUpPassword, TestData.VALID_SIGNUP_PASSWORD);
```
Done

francoisa-inga commented on akopova/src/main/java/pages/ParentPage.java:
```diff
@@ -109,6 +111,25 @@ protected void checkIsElementVisible(WebElement webElement){
+        , isElementEnabled(webElement));
+    }
+
+    protected boolean isElementEnabled(WebElement webElement) {        try{
```
Поправила. Это я попробовала еще проверить кнопку SignUp, if Enabled. Думала может поможет чтоб она кликалась, но не помогло
